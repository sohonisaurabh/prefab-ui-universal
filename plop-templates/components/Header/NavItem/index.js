/**@jsx jsx */
import React from "react";
import PropTypes from "prop-types";
import { jsx, css } from "@emotion/core";
import Link from "next/link";
const navItemStyle = css`
  color: #fff;
  font-size: 1.3em;
  font-weight: bold;
  text-transform: capitalize;
  text-decoration: none;
  width: 100px;
  height: 18px;
  margin-right: 20px;
  cursor: pointer;
  transition: all 0.3s;
`;

const NavItem = ({ index, active, linkName, onClick }) => (
  <React.Fragment>
    <Link href={linkName.url}>
      <a css={navItemStyle}> {linkName.title}</a>
    </Link>
    {/* <a href={linkName.url} onClick={() => onClick(index)} css={navItem}>
      {linkName.title}
    </a> */}
  </React.Fragment>
);

NavItem.propTypes = {
  active: PropTypes.bool,
  index: PropTypes.number.isRequired,
  onClick: PropTypes.func
};

NavItem.defaultProps = {
  active: false,
  onClick: () => {}
};

export default NavItem;
