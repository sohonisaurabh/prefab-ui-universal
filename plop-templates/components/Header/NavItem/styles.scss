.nav-item {
  color: #fff;
  font-size: 1.3em;
  font-weight: bold;
  text-transform: capitalize;
  text-decoration: none;
  width: 100px;
  height: 18px;
  margin-right: 20px;
  cursor: pointer;
  transition: all 0.3s;
}
