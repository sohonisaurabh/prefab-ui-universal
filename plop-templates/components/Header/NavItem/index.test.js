import React from 'react';
import renderer from 'react-test-renderer';
import NavItem from './index';

it('NavItem renders correctly', () => {
  const tree = renderer.create(<NavItem />).toJSON();
  expect(tree).toMatchSnapshot();
});
