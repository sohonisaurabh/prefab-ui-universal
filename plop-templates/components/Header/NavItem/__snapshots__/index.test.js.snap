// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem renders correctly 1`] = `
<a
  className=" css-5yvc01-NavItem"
  onClick={[Function]}
/>
`;
